### PR TITLE
Onboarding : Machine-assisted article descriptions

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T331272

- Open up the onboarding screens to **all** desc editing, not just when entering via an article.
- The 3rd screen on Description editing onboarding is shown only to users who will see the suggestions among the AB groups.

### CODE REVIEW

**Comment by sharvaniharan** [comment]
This is ready @cooltey @dbrant

File: app/src/main/java/org/wikipedia/descriptions/DescriptionEditActivity.kt [hunk 1]
```diff
@@ -2,29 +2,43 @@ package org.wikipedia.descriptions
...
+        val action = intent.getSerializableExtra(Constants.INTENT_EXTRA_ACTION) as Action
+        if (action == Action.ADD_DESCRIPTION && Prefs.isDescriptionEditTutorialEnabled) {
+            Prefs.isDescriptionEditTutorialEnabled = false
+            val pageTitle = intent.getParcelableExtra<PageTitle>(EXTRA_TITLE)!!
+            val shouldShowAIOnBoarding = ReleaseUtil.isPreBetaRelease && SuggestedArticleDescriptionsDialog.availableLanguages()
+                .contains(pageTitle.wikiSite.languageCode) && pageTitle.description.isNullOrEmpty() &&
+                    WikipediaApp.instance.machineGeneratedDescriptionsABTest.aBTestGroup != GROUP_1
+            startActivity(DescriptionEditTutorialActivity.newIntent(this, shouldShowAIOnBoarding))
+        }
+    }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/descriptions/DescriptionEditActivity.kt:41]: Please add an empty line here.

File: app/src/main/java/org/wikipedia/descriptions/DescriptionEditTutorialFragment.kt [hunk 2]
```diff
@@ -42,9 +43,10 @@ class DescriptionEditTutorialFragment : OnboardingFragment() {
     companion object {
         const val ARG_POSITION = "position"
         val pages = arrayOf(R.layout.inflate_description_edit_tutorial_page_one, R.layout.inflate_description_edit_tutorial_page_two)
+        val AiIncludedPages = arrayOf(R.layout.inflate_description_edit_tutorial_page_one, R.layout.inflate_description_edit_tutorial_page_two, R.layout.inflate_description_edit_tutorial_page_three)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/descriptions/DescriptionEditTutorialFragment.kt:46]: Would it be possible to make the `pages` includes all layouts so we can handle the logic only in `getItemCount()`?

```
       override fun getItemCount(): Int {
            return if (showAIOnBoarding) pages.size else pages.size - 1
        }
```

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/descriptions/DescriptionEditTutorialFragment.kt (hunk 2 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/descriptions/DescriptionEditTutorialFragment.kt]: Good idea!

**Review by cooltey — CHANGES_REQUESTED** [review]
Looks like the size of the image is not correct compares to the one in ticket.

**Review by cooltey — CHANGES_REQUESTED** [review]
Found another issue: if the user change device orientation while view onboarding screen, the app will crash.

**Comment by sharvaniharan** [comment]
Done @cooltey

**Pull request closed without merging**